### PR TITLE
Ensure that no special characters are included in the drupal username

### PR DESCRIPTION
**Pull request by seamuslee001**

### CODE REVIEW

File: aoservicelisting.civix.php [hunk 1]
```diff
@@ -305,14 +305,26 @@ public static function emailRule($cid) {
...
+    return $nameDetails['first_name'] . '.' . $nameDetails['last_name'] . $nameDetails['id'];
+  }
 
   public static function createUserAccount($cid) {
     // We create the user account for the primary contact.
     if (empty($cid)) {
       return FALSE;
     }
-    $name = CRM_Core_DAO::executeQuery("SELECT LOWER(CONCAT(first_name, '.', last_name, $cid)) AS name, display_name
+    $name = CRM_Core_DAO::executeQuery("SELECT id, first_name, last_name AS name, display_name
```
Comment (monishdeb) [inline aoservicelisting.civix.php]: @seamuslee001 please remove the alias of last_name, also display_name is not needed now

**Review by Edzelopez — APPROVED** [review]
Patch looks good.

File: aoservicelisting.civix.php (hunk 1 above)
Comment (seamuslee001) [reply aoservicelisting.civix.php]: @monishdeb display_name is still used in L342

Comment (monishdeb) [reply aoservicelisting.civix.php]: oh ok my bad, yup its fine then to leave display_name

**Comment by seamuslee001** [comment]
ok updated PR after doing some testing on staging

**Pull request merged**